gubernator/main.py

```diff
@@ -87,10 +52,38 @@ class Warmup(webapp2.RequestHandler):
     """Warms up gubernator."""
     def get(self):
         """Receives the warmup request."""
-        # TODO(fejta): warmup something useful
+        self.app.config['github_client'] = secrets.get('github_client')
+
         self.response.headers['Content-Type'] = 'text/plain'
         self.response.write('Warmup successful')
 
+
+class ConfigHandler(view_base.BaseHandler):
+    """Handles admin config of gubernator."""
+    def get(self):
+        self.render('config.html', {'hostname': hostname})
+
+    def post(self):
+        self.response.headers['Content-Type'] = 'text/plain'
+        if users.is_current_user_admin():
+            github_id = self.request.get('github_id')
+            github_secret = self.request.get('github_secret')
+            if github_id and github_secret:
+                value = {'id': github_id, 'secret': github_secret}
+                secrets.put('github_client', value)
+                app.config['github_client'] = value
+                self.response.write('set github oauth client!\n')
+            github_webhook_secret = self.request.get('github_webhook_secret')
+            if github_webhook_secret:
+                secrets.put('github_webhook_secret',
+                            github_webhook_secret,
+                            per_host=False)
+                self.response.write('set github webhook secret!\n')
+            self.response.write('done.')
+        else:
+            self.response.write('not admin, ignoring.')
```

Comment:
I think `fejta@google.com is not an admin.` might be more helpful here. Also return a 401/403?

Reply:
you can't even view the page if you're not an admin, so I'm not going to improve POST error messages.